modules/framework/galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/internal/streams/Stream.java

```diff
@@ -32,22 +41,33 @@ public void setDescription(String description) {
         this.description = description;
     }
 
-    public String getMavenRepositoryUrl(){
+    @Override
+    public URL getMavenRepositoryUrl(){
         return this.mavenRepositoryUrl;
     }
 
-    public void setMavenRepositoryUrl(String mavenRepositoryUrl) {
-        this.mavenRepositoryUrl = mavenRepositoryUrl;
+    public void setMavenRepositoryUrl(String mavenRepositoryUrl) throws StreamsException {
+        try {
+            this.mavenRepositoryUrl = new URL(mavenRepositoryUrl);
+        } catch (MalformedURLException e) {
+            throw new StreamsException("Invalid maven repository URL provided", e);
+        }
     }
 
-    public String getTestCatalogUrl() {
+    @Override
+    public URL getTestCatalogUrl() {
         return this.testCatalogUrl;
     }
 
-    public void setTestCatalogUrl(String testCatalogUrl) {
-        this.testCatalogUrl = testCatalogUrl;
+    public void setTestCatalogUrl(String testCatalogUrl) throws StreamsException {
+        try {
+            this.testCatalogUrl = new URL(testCatalogUrl);
+        } catch (MalformedURLException e) {
+            throw new StreamsException("Invalid testcatalog URL provided", e);
+        }
     }
 
+    @Override
     public boolean getIsEnabled() {
```

Comment:
isEnabled ??

Reply:
Addressed in https://github.com/galasa-dev/galasa/pull/236/commits/a9ef197a47c1c7da97f4c1566f80ca24bce91184